packages/odyssey-react/src/components/Tab/index.tsx

```diff
@@ -106,9 +109,51 @@ const Tabs: FunctionComponent<PropsTabs> & StaticComponents = ({ children, id, s
     }
   }
 
+  const handleTabFocus = (event: KeyboardEvent) => {
+    if (tabElement.current) {
+      const focusableSelector = `
+      a[href],
+      button,
+      textarea,
+      input[type="text"],
+      input[type="radio"],
+      input[type="checkbox"],
+      select
+      `;
+      const tabs = tabElement.current.querySelectorAll(focusableSelector);
+      const tabsArr = Array.prototype.slice.call(tabs);
+      const tabCount = tabs.length;
+      const focusableElFirst = tabs[0] as HTMLElement;
+      const focusableElLast = tabs[tabCount - 1] as HTMLElement;
+      const focusableIndexActive = tabsArr.indexOf(document.activeElement);
+      const focusableIndexLast = tabsArr.indexOf(focusableElLast);
+      const focusableElNext = tabs[focusableIndexActive + 1] as HTMLElement;
+      const focusableElPrev = tabs[focusableIndexActive - 1] as HTMLElement;
+      const isFirstFocused = focusableIndexActive === 0;
+      const isLastFocused = focusableIndexActive === focusableIndexLast
+      const focusFirst = () => focusableElFirst.focus();
+      const focusLast = () => focusableElLast.focus();
+      const keyPressed = (key: string) => key === event.key;
+
+      if (keyPressed('Home')) focusFirst();
+      else if (keyPressed('End')) focusLast();
+      else if (keyPressed('ArrowLeft') && !isFirstFocused) focusableElPrev.focus();
+      else if (keyPressed('ArrowLeft') && isFirstFocused) focusLast();
+      else if (keyPressed('ArrowRight') && !isLastFocused) focusableElNext.focus();
+      else if (keyPressed('ArrowRight') && isLastFocused) focusFirst();
+    }
+  }
```

Comment:
can we extract this into a well named function or two? kinda hard to parse. The selector also doesn't need to be redefined on every render pass